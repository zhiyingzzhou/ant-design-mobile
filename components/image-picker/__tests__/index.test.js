import React from 'react';
import { render, mount, shallow } from 'enzyme';
import Flex from '../../flex';
import ImagePicker from '../index';

describe('ImagePicker', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <ImagePicker />,
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.html()).toEqual('<div class="am-image-picker-list" role="group"><div class="am-flexbox am-flexbox-align-center"><div class="am-flexbox-item"><div class="am-image-picker-item am-image-picker-upload-btn" role="button" aria-label="Choose and add image"><input type="file" accept="image/*"></div></div><div class="am-flexbox-item"></div><div class="am-flexbox-item"></div><div class="am-flexbox-item"></div></div></div>');
  });

  it('renders files correctly', () => {
    const files = [{
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121',
    }, {
      url: 'https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg',
      id: '2122',
    }];

    const wrapperCant = mount(
      <ImagePicker
        selectable={false}
        files={files}
      />,
    );
    // can't select
    expect(wrapperCant.find('.am-image-picker-item').length).toBe(2);

    const wrapper = mount(
      <ImagePicker
        files={files}
      />,
    );
    // can select
    expect(wrapper.find('.am-image-picker-item').length).toBe(3);
  });

  it('select file correctly', () => {
    const wrapperCant = mount(
      <ImagePicker
        selectable={false}
      />,
    );
    // can select
    expect(wrapperCant.find('.am-image-picker-upload-btn').length).toBe(0);

    const wrapper = mount(
      <ImagePicker />,
    );
    // can select
    expect(wrapper.find('.am-image-picker-upload-btn').length).toBe(1);
  });

  it('renders length', () => {
    const files = [{
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121',
    }, {
      url: 'https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg',
      id: '2122',
    }, {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2123',
    }];

    const wrapperDefault = shallow(
      <ImagePicker
        files={files}
        selectable={false}
      />,
    );
    expect(wrapperDefault.find(Flex).length).toBe(1);

    const wrapperWithTwo = shallow(
      <ImagePicker
        length="2"
        files={files}
        selectable={false}
      />,
    );
    expect(wrapperWithTwo.find(Flex).length).toBe(2);

    const wrapperWithOne = shallow(
      <ImagePicker
        length="1"
        files={files}
        selectable={false}
      />,
    );
    expect(wrapperWithOne.find(Flex).length).toBe(3);
  });
});
