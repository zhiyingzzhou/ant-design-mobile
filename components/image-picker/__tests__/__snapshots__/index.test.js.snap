// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePicker renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "am-image-picker",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "am-image-picker-list",
          "role": "group",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "am-flexbox am-flexbox-align-center",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "am-flexbox-item",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "aria-label": "Choose and add image",
                      "class": "am-image-picker-item am-image-picker-upload-btn",
                      "role": "button",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "accept": "image/*",
                          "type": "file",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "accept": undefined,
                          "type": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "accept": undefined,
                          "type": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-label": undefined,
                      "class": undefined,
                      "role": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-label": undefined,
                      "class": undefined,
                      "role": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "am-flexbox-item",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "am-flexbox-item",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "class": "am-flexbox-item",
                      },
                      "children": Array [],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "am-flexbox-item",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "am-flexbox-item",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "am-flexbox-item",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "am-flexbox-item",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "aria-label": "Choose and add image",
                        "class": "am-image-picker-item am-image-picker-upload-btn",
                        "role": "button",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "accept": "image/*",
                            "type": "file",
                          },
                          "children": Array [],
                          "name": "input",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "accept": undefined,
                            "type": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "accept": undefined,
                            "type": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "aria-label": undefined,
                        "class": undefined,
                        "role": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "aria-label": undefined,
                        "class": undefined,
                        "role": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "am-flexbox-item",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "am-flexbox-item",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "am-flexbox-item",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "am-flexbox-item",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "aria-label": "Choose and add image",
                          "class": "am-image-picker-item am-image-picker-upload-btn",
                          "role": "button",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "accept": "image/*",
                              "type": "file",
                            },
                            "children": Array [],
                            "name": "input",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "accept": undefined,
                              "type": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "accept": undefined,
                              "type": undefined,
                            },
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "aria-label": undefined,
                          "class": undefined,
                          "role": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "aria-label": undefined,
                          "class": undefined,
                          "role": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "am-flexbox-item",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "am-flexbox-item",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "am-flexbox-item",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "class": "am-flexbox-item",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "aria-label": "Choose and add image",
                            "class": "am-image-picker-item am-image-picker-upload-btn",
                            "role": "button",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "accept": "image/*",
                                "type": "file",
                              },
                              "children": Array [],
                              "name": "input",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "accept": undefined,
                                "type": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "accept": undefined,
                                "type": undefined,
                              },
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "aria-label": undefined,
                            "class": undefined,
                            "role": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "aria-label": undefined,
                            "class": undefined,
                            "role": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
